src/Microsoft.DotNet.ProjectModel/Graph/BuildAction.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.ProjectModel.Graph
+{
+    public struct BuildAction : IEquatable<BuildAction>
+    {
+        public static readonly BuildAction Compile = new BuildAction(nameof(Compile));
+        public static readonly BuildAction EmbeddedResource = new BuildAction(nameof(EmbeddedResource));
+        public static readonly BuildAction Resource = new BuildAction(nameof(Resource));
+
+        // Default value
+        public static readonly BuildAction None = new BuildAction(nameof(None));
+
+        public string Value { get; }
+
+        private BuildAction(string value)
+        {
+            Value = value;
+        }
+
+        public static bool TryParse(string value, out BuildAction type)
```

Comment:
None is valid build action now, I'd like to keep it consistent with LibraryType code and keep TryParse